### DISCUSSION TITLE
Monthly updates missing schedule for onboarded repositories

### DISCUSSION THREAD

**Discussion by jaroslavhavelik** [renovatebot/renovate#32268] [Request Help]
### How are you running Renovate?

A Mend.io-hosted app

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

BItbucket app

### Please tell us more about your question or problem

We are using mend.io-hosted app for the Bitbucket cloud. For several repositories, we have the following schedule config:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "baseBranches": [
    "dev"
  ],
  "hostRules": [
    {
      "matchHost": "api.github.com",
      "token": "{{ secrets.MEND_GITHUB_COM_TOKEN }}"
    }
  ],
  "prHourlyLimit": 0,
  "prConcurrentLimit": 0,
  "schedule": [
    "before 4am on the first day of the month"
  ]
}

```

For some reason several of our app with this config are in onboarded status (running daily) and missing schedule, instead of active (running hourly). So PRs are not created as they are missing schedules.

Is there a way how to move repo from onboarded status to active?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Found PR #112
DEBUG: branchExists=false
DEBUG: dependencyDashboardCheck=undefined
DEBUG: PR rebase requested=false
DEBUG: Checking schedule(schedule=before 4am on the first day of the month, tz=null, now=2024-10-31T02:44:58.351Z)
DEBUG: Checking 1 schedule(s)
DEBUG: Checking schedule "before 4am on the first day of the month"
DEBUG: Package not scheduled
DEBUG: Skipping branch creation as not within schedule

```

</details>

**Comment by rarkins** [comment]
Mend-hosted apps schedule "activated" repos hourly, and "onboarded" repos daily. An activated repo is one which has merged at least one Renovate PR.

**Reply by jaroslavhavelik** [reply]
@rarkins okay, that makes sense. We have definitely multiple Renovate PRs merged within this repo, but it's still marked as onboarded, any suggestions what we should do to doublecheck?

**Reply by rarkins** [reply]
The 'activated' status is as reported by the Renovate CLI. There's a debug message starting with ``Repository result:` you can find in every run and see what it says. Start with one repo which shows up as onboarded in the portal when you think it should be activated.

**Reply by jaroslavhavelik** [reply]
In this specific repo message says: `Repository result: repository-changed, status: onboarded, enabled: true, onboarded: true`.

Can a change in `renovate.json` cause this? We did it during September (for example changing the schedule).

**Reply by rarkins** [reply]
No, it's caused if Renovate fetches the list of PRs and finds no merged ones (other than the Onboarding PR). You're welcome to submit a reproduction repo if you think it's not working as expected

**Reply by jaroslavhavelik** [reply]
@rarkins I've tried to create a reproduction repository but it was successfully onboarded and activated. I found one difference within my original repo and reproduction. Even when renovate detected updates, it doesn't offer me creation of updates for this branch, see the screenshot:

![image](https://github.com/user-attachments/assets/cd5bb0a8-a73f-4cd3-a720-0bd5e4e8a174)

So I'm in sort of a deadlock, as I'm not able to create renovate PRs manually and PRs won't be created automatically as they probably miss the schedule.
Still not sure why Renovate thinks that no PR is merged (repo is not activated) when I see several merged PRs:

![image](https://github.com/user-attachments/assets/3302f3e8-7edc-46a2-b2d8-6c454aadedd5)

**Reply by rarkins** [reply]
Please share the full debug log of one such repository which has status=onboarded where you expect status=activated

**Reply by jaroslavhavelik** [reply]
Attached here: [renovate_log.txt](https://github.com/user-attachments/files/17629432/renovate_log.txt)

**Reply by rarkins** [reply]
Is this expected?

```
DEBUG: Found existing branch PR (branch="renovate/com.appsamurai.storyly-storyly-3.x")
{
  "baseBranch": "dev"
}

DEBUG: PR has been closed or merged since this run started - aborting (branch="renovate/com.appsamurai.storyly-storyly-3.x")
{
  "baseBranch": "dev"
}

DEBUG: Passing repository-changed error up (branch="renovate/com.appsamurai.storyly-storyly-3.x")
{
  "baseBranch": "dev"
}

INFO: Repository has changed during renovation - aborting
DEBUG: Repository result: repository-changed, status: onboarded, enabled: true, onboarded: true
```

Renovate aborted that run, thinking that the PR was closed/merge since its start.

I have a suspicion that there could be an empty PR list, which would also explain why Renovate doesn't consider the repo to be activated

**Reply by rarkins** [reply]
I added https://github.com/renovatebot/renovate/issues/32329 for @zharinov to look into soon

Adding logging verbosity to the getPrList() in BB should help us uncover the bug or realize what's wrong with this repo.

**Reply by jaroslavhavelik** [reply]
```
DEBUG: Found existing branch PR (branch="renovate/com.appsamurai.storyly-storyly-3.x")
{
  "baseBranch": "dev"
}

DEBUG: PR has been closed or merged since this run started - aborting (branch="renovate/com.appsamurai.storyly-storyly-3.x")
{
  "baseBranch": "dev"
}

DEBUG: Passing repository-changed error up (branch="renovate/com.appsamurai.storyly-storyly-3.x")
{
  "baseBranch": "dev"
}
```

I would say it is expected that there are closed PRs with this change. However, this update had a breaking change, so we had to add one more commit to the branch.

@rarkins, I'm not sure what you mean by an empty PR list. There are no open PRs in this repository at the moment, but there are several closed ones.

Let me know when you deploy this change I'll rerun the scan and attach new logs

**Reply by rarkins** [reply]
The PR list retrieves both open and closed.

BTW for other repos which don't have this same problem, do they also have baseBranches configured?

**Reply by jaroslavhavelik** [reply]
Yes, it shouldn't be the case, our activated repos have also `baseBranches` set.

**Reply by jaroslavhavelik** [reply]
@rarkins I think we've found the root cause of this. It seems that if we have a branch in the repository with an update (`renovate/.....`) that doesn't have an open PR (for example, when a PR was autoclosed by a different PR), the repository is, for some reason, only onboarded and even does not offer creating/rebasing PRs via the developer.mend.io console. After we deleted all of these branches, the repository became active and offered PRs via the console.

Is this expected behavior or some sort of an issue?

**Reply by rarkins** [reply]
No, please feel free to contribute a fix or reproduction
